src/drive/web/modules/viewer/Footer/BottomSheet.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { BottomSheet } from 'mui-bottom-sheet'
+import { makeStyles } from '@material-ui/core/styles'
+
+const useStyles = ({ isTopPosition }) => ({
+  root: {
+    borderTopLeftRadius: '1rem',
+    borderTopRightRadius: '1rem',
+    transition: 'border-radius 0.5s',
+    boxShadow: '0 6px 16px 0 rgba(0, 0, 0, 0.5)',
+    ...(isTopPosition && {
+      borderTopLeftRadius: 0,
+      borderTopRightRadius: 0
+    })
+  }
+})
+
+const useClasses = makeStyles(theme => ({
+  header: {
+    height: '2.5rem',
+    width: '100%',
+    position: 'relative',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center'
+  },
+  indicator: {
+    width: '4rem',
+    height: '0.25rem',
+    borderRadius: '99px',
+    backgroundColor: theme.palette.divider
+  }
+}))
+
+const BottomSheetWrapper = ({ file, actionButtonsRef, children }) => {
+  const [isTopPosition, setIsTopPosition] = useState(false)
+  const [peekHeights, setPeekHeights] = useState(null)
+  const [initPos, setInitPos] = useState(null)
+  const [bottomSpacerHeight, setBottomSpacerHeight] = useState(0)
+  const classes = useClasses()
+  const styles = useStyles({ isTopPosition })
+  const innerContentRef = useRef()
+  const headerRef = useRef()
+
+  const toolbar = document.getElementById('viewer-toolbar')
```

Comment:
No. Please, do not access any DOM APIs inside render, please use a ref or a useMemo.

Reply:
Ok pour `useMemo` cf https://github.com/cozy/cozy-ui/pull/1769#discussion_r594441262